frontend/src/Editor/ReleaseVersionButton.jsx

```diff
@@ -9,11 +9,13 @@ export const ReleaseVersionButton = function DeployVersionButton({
   isVersionReleased,
   fetchApp,
   onVersionRelease,
+  saveEditingVersion,
 }) {
   const [isReleasing, setIsReleasing] = useState(false);
 
   const releaseVersion = (editingVersion) => {
     setIsReleasing(true);
+    saveEditingVersion();
```

Comment:
Should there be an `await` ?

Reply:
Cant make use await here, as that function comprises of setState calls which don't return promises.
We would have to change this up to handle with multiple callbacks to ensure it runs serially. 

But if I'm not wrong having this run concurrently is fine as one API call is to update the current definition on the canvas and the other to update the current version id and there isn't any dependency on their order of execution.